psse/psse-converter/src/main/java/com/powsybl/psse/converter/PsseImporter.java

```diff
@@ -0,0 +1,358 @@
+    public Network importData(ReadOnlyDataSource dataSource, NetworkFactory networkFactory, Properties parameters) {
+        Objects.requireNonNull(dataSource);
+        Objects.requireNonNull(networkFactory);
+
+        Network network = networkFactory.createNetwork(dataSource.getBaseName(), FORMAT);
+
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(dataSource.newInputStream(null, EXT)))) {
+
+            // parse file
+            PsseRawModel psseModel = new PsseRawReader().read(reader);
+
+            // set date and time
+            // TO DO
+
+            // build container to fit IIDM requirements
+            List<Object> branches = ImmutableList.builder()
+                    .addAll(psseModel.getNonTransformerBranches())
+                    .addAll(psseModel.getTransformers())
+                    .build();
+            ToIntFunction<Object> branchToNum1 = branch -> branch instanceof PsseNonTransformerBranch ? ((PsseNonTransformerBranch) branch).getI() : ((PsseTransformer) branch).getFirstRecord().getI();
+            ToIntFunction<Object> branchToNum2 = branch -> branch instanceof PsseNonTransformerBranch ? ((PsseNonTransformerBranch) branch).getJ() : ((PsseTransformer) branch).getFirstRecord().getJ();
+            ToDoubleFunction<Object> branchToResistance = branch -> branch instanceof PsseNonTransformerBranch ? ((PsseNonTransformerBranch) branch).getR() : ((PsseTransformer) branch).getSecondRecord().getR12();
+            ToDoubleFunction<Object> branchToReactance = branch -> branch instanceof PsseNonTransformerBranch ? ((PsseNonTransformerBranch) branch).getX() : ((PsseTransformer) branch).getSecondRecord().getX12();
+            Predicate<Object> branchToIsTransformer = branch -> branch instanceof PsseTransformer;
+            ContainersMapping containerMapping = ContainersMapping.create(psseModel.getBuses(), branches, PsseBus::getI, branchToNum1,
+                                                                          branchToNum2, branchToResistance, branchToReactance, branchToIsTransformer);
+
+            boolean ignoreBaseVoltage = ConversionParameters.readBooleanParameter(FORMAT, parameters, IGNORE_BASE_VOLTAGE_PARAMETER,
+                    ParameterDefaultValueConfig.INSTANCE);
+            PerUnitContext perUnitContext = new PerUnitContext(psseModel.getCaseIdentification().getSbase(), ignoreBaseVoltage);
+
+            // create buses
+            createBuses(psseModel, containerMapping, perUnitContext, network);
+
+            //Create loads
+            for (PsseLoad psseLoad : psseModel.getLoads()) {
+                createLoad(psseLoad, network.getVoltageLevel(containerMapping.getVoltageLevelId(psseLoad.getI())));
+            }
+
+            //Create shunts
+            for (PsseFixedShunt psseShunt : psseModel.getFixedShunts()) {
+                createShuntCompensator(psseShunt, perUnitContext, network.getVoltageLevel(containerMapping.getVoltageLevelId(psseShunt.getI())));
```

Comment:
I would pass the network and containerMapping to createShuntCompensator and get the voltage level inside.